projects/frontend/data-pipelines/gui/projects/ui/src/app/app-config.service.ts

```diff
@@ -8,22 +8,28 @@ import { HttpBackend, HttpClient } from '@angular/common/http';
 import { AuthConfig } from 'angular-oauth2-oidc';
 import { AppConfig, RefreshTokenConfig } from './app-config.model';
 import { firstValueFrom } from 'rxjs';
-
+import { Router } from '@angular/router';
+import { routes } from './app.routing';
 @Injectable({
     providedIn: 'root'
 })
 export class AppConfigService {
     private httpClient: HttpClient;
     private appConfig: AppConfig;
+    private router: Router;
 
-    constructor(private httpBackend: HttpBackend) {
+    constructor(private httpBackend: HttpBackend, private angularRouter: Router) {
         this.httpClient = new HttpClient(httpBackend);
+        this.router = angularRouter;
     }
 
-    loadAppConfig(): Promise<void> {
-        return firstValueFrom(this.httpClient.get<AppConfig>('/assets/data/appConfig.json').pipe()).then((data) => {
-            this.appConfig = data;
+    async loadAppConfig(): Promise<void> {
```

Comment:
It's the same thing we were doing before, just better syntax, imo. We need to return a promise. If you call the `async` method without `await`, it will not wait for the promise to resolve and just return it. This is what we want for the APP_Initializer as per https://angular.io/api/core/APP_INITIALIZER

Reply:
I think chaining the promises would give better flow control.
e.g.
```
loadAppConfig(): Promise<boolean> {
    const data = await firstValueFrom(this.httpClient.get<AppConfig>( '/assets/data/appConfig.json').pipe());
        this.appConfig = data;
        const localRoutes = routes.filter((route: { path: string }) => {
            return !data.ignoreRoutes.includes(route.path);
        });
        this.router.resetConfig(localRoutes);

        return firstValueFrom(this.httpClient.get<AppConfig>( '/assets/data/appConfig.json'))
                          .then((data) => {
                                  this.appConfig = data;

                                  try {
                                        const localRoutes = routes.filter((route: { path: string }) => {
                                                   return !data.ignoreRoutes.includes(route.path);
                                        });

                                        this.router.resetConfig(localRoutes);

                                        return true;
                                  } catch(e) {
                                         console.error('Failed to reset Router config');

                                         // when exception occurs you can either re-throw the exception and stop application bootstrap
                                         throw e;

                                         // or return boolean true and continue with application bootstrap and default config
                                         // return true;
                                  }
                          });
}
```

Reply:
Done.